### ISSUE TITLE
Search/Filter Suggestion

### ISSUE THREAD

**Issue by chinagreenelvis** [nptr/msstyleEditor#23]
Would be great to be able to search for things like "textcolor" or "fillcolor", or a filter to better find those things without knowing the exact RGB.

Also, an "expand all" ability for the left-hand list?

**Comment by nptr** [comment]
I agree, some kind of fuzzy search would be nice. If you have some specific idea how it could looks like
or know an example, let me know.

While it's not on top of my list, maybe i can add some short-term improvements for the next release.

**Comment by TheArkive** [comment]
I'd like to suggest an editbox/combobox at the top or bottom of the left panel.  A search history would be cool to for quick recall.  Also being able to specify searching of classes, parts, properties in any combination would also be cool, perhaps in a right-click context menu or dropdown menu?

**Comment by TheArkive** [comment]
Here's an example of what I would like to see in the filter's style:
https://i.imgur.com/JOQeJl4.png

I'd like to see it at the top of the left panel listing all the classes.

Add to this a right-click context menu on the combo box right-click event to select the scope of the search, such as "classes", "parts", "states", and "properties".

Activating the search can either be a time delay, or require the ENTER key.  Once the search is activated, the contents of the left pane tree list are refreshed to only match the search query.

Here's how I'd like to see it function:

1) When filtering classes, a search of "address" would list all entries that have the word "address" in it.  Then the left panel tree list would refresh.  Hopefully the list of classes will stay in memory so the file doesn't have to be parsed again for all the filtered class lists.  Classes formatted as CLASS::INHERITED should still be formatted as such.  I think fully knowing exactly what is inherited is not only quite helpful, but I like the way it is already displayed in the left panel tree list.  Short and to the point.

2) When searching parts and states, the same as 1 above, but expand the tree for all items.  Omit all parts and states that are not matched, or nested with the search query in the left panel tree.  Or, simply highlight the matched parts (via text color change, red??) and states and don't omit the rest of the nested parts and states that are not matched by the search.  I could go either way on that.

3) When searching properties, exactly the same as 1 & 2 above, but when selecting a part or state that has properties, simply "highlight" the properties (via text color change again) in the right panel (red?  blue?  ... i donno).  Just changing the text of the matched property would be a short and simple way to point out the matching property.

So a lot of what I'm getting at here is actually very similar to how WSB does it's search.  The only difference is that you aren't trying to categorize every single class the way WSB does.  That adds an extra layer that I personally don't need.  Just let the class list be what it is.

Hope that helps.  If you would like more pics / screenshots for more visuals I can do that.
